packages/m2m-gateway/src/services/agreementService.ts

```diff
@@ -196,6 +196,22 @@ export function agreementServiceBuilder(clients: PagoPAInteropBeClients) {
       );
       const polledResource = await pollAgreement(response, headers);
 
+      return toM2MGatewayApiAgreement(polledResource.data);
+    },
+    async upgradeAgreement(
+      agreementId: AgreementId,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.Agreement> {
+      logger.info(`Upgrading agreement with id ${agreementId}`);
+
+      const response =
+        await clients.agreementProcessClient.upgradeAgreementById(undefined, {
```

Comment:
```suggestion
        await clients.agreementProcessClient.upgradeAgreement(undefined, {
```

I just noticed that, in the process, the `upgradeAgreement` and `updateAgreement` have the suffix `ById`, while all the other operations don't.
I think we could align the naming of these two operations with the others in this PR as well, it should be enough to replace the `operationId` in the `agreementApi` and replace the two wherever they're used (I think they're being directly used only in the BFF).